CanDIG/query
# DIG-1240: API middleware service to handle queries across federated instances

# Issues
[DIG-1240](https://candig.atlassian.net/browse/DIG-1240)

# Description
This implements a new query microservice, which is local to each instance.

# To Test:
- Spin up the entire stack, including the query microservice commits in both candig-data-portal and the main CanDIGv2 repo.
- The query microserivce has two endpoints: `candig.docker.internal:1236/service-info` and `candig.docker.internal:1236/query`
- The federation fanout query should work if you hit: `{ "path": "query", "service": "query", "method": "GET" }`
- The CanDIG data portal should work fine with the new query microservice
- Query parameters are: 
  - treatment
  - primarySite
  - chemotherapy
  - immunotherapy
  - hormoneTherapy
  - chr
  - gene
  - assembly
  - pageSize
  - pageNum
(For more details, see the OpenAPI spec [here](https://github.com/CanDIG/query/blob/71ac47a1d4d4de70ab85589ff8a531b87dcd7b97/query_server/openapi.yaml)

[DIG-1240]: https://candig.atlassian.net/browse/DIG-1240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
